From: Marco Nenciarini <marco.nenciarini@2ndquadrant.it>
Date: Wed, 4 Dec 2019 18:15:28 +0100
Subject: Avoid connecting to PostgreSQL while reading a backup.info file

Prevent Barman from opening a PostgreSQL transaction that lasts until
termination, even in operations that should not require to speak with
Postgres (i.e. list-backup or recover).

This fixes a regression introduced with systemid check code.

https://github.com/2ndquadrant-it/barman/commit/dde2d120a17c243392e65d53e4f3f0ae7784b742
---
 barman/backup.py   | 1 +
 barman/infofile.py | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/barman/backup.py b/barman/backup.py
index bfea880..21e4aa2 100644
--- a/barman/backup.py
+++ b/barman/backup.py
@@ -390,6 +390,7 @@ class BackupManager(RemoteStatusMixin):
             backup_info = LocalBackupInfo(
                 self.server,
                 backup_id=datetime.datetime.now().strftime('%Y%m%dT%H%M%S'))
+            backup_info.set_attribute('systemid', self.server.systemid)
             backup_info.save()
             self.backup_cache_add(backup_info)
             output.info(
diff --git a/barman/infofile.py b/barman/infofile.py
index 03912c8..ffed6f4 100644
--- a/barman/infofile.py
+++ b/barman/infofile.py
@@ -596,7 +596,6 @@ class LocalBackupInfo(BackupInfo):
                     'both info_file and backup_id parameters are set')
             self.backup_id = backup_id
             self.filename = self.get_filename()
-            self.systemid = server.systemid
             # Check if a backup info file for a given server and a given ID
             # already exists. If so load the values from the file.
             if os.path.exists(self.filename):
